hail/src/main/scala/is/hail/expr/types/physical/PCanonicalArray.scala

```diff
@@ -367,4 +375,82 @@ final case class PCanonicalArray(elementType: PType, required: Boolean = false)
       destOff
     )
   }
+
+  override def storeShallowAtOffset(dstAddress: Code[Long], valueAddress: Code[Long]): Code[Unit] =
+    Region.storeAddress(dstAddress, valueAddress)
+
+  override def storeShallowAtOffset(dstAddress: Long, valueAddress: Long) {
+    Region.storeAddress(dstAddress, valueAddress)
+  }
+
+  // semantically this function expects a non-null sourceAddress, and by that property, this function results in a non-null value
```

Comment:
This doesn't seem like a well-defined clarification in our system, since addresses cannot be null. It's like saying:

```scala
// this function expects a non-null value for 'i'
def doubleIt(i: Int): Int = i * 2
```

Reply:
I was thinking about null pointers at this time, is this point worth clarifying?